### PR TITLE
Pv2 534 create datalock projects

### PR DESCRIPTION
**Pull request by samuel-dev-dba**

### CODE REVIEW

File: src/SFA.DAS.Payments.Datalocks.DatalocksService.Interfaces/IDatalocksService.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+using Microsoft.ServiceFabric.Actors;
+using Microsoft.ServiceFabric.Actors.Remoting.FabricTransport;
+using Microsoft.ServiceFabric.Services.Remoting;
+
+[assembly: FabricTransportActorRemotingProvider(RemotingListenerVersion = RemotingListenerVersion.V2_1, RemotingClientVersion = RemotingClientVersion.V2_1)]
+namespace SFA.DAS.Payments.Datalocks.DatalocksService.Interfaces
+{
+    public interface IDatalocksService : IActor
+    {
+        Task<int> GetCountAsync(CancellationToken cancellationToken);
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.Datalocks.DatalocksService.Interfaces/IDatalocksService.cs]: The ticket suggests that event classes are to be created as a part of this work. This interface needs a method for handling earning events and returning newliy defined payable earning events.

File: src/SFA.DAS.Payments.Datalocks.DatalocksService.Interfaces/SFA.DAS.Payments.Datalocks.DatalocksService.Interfaces.csproj [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<Project Sdk="Microsoft.NET.Sdk">
```
[the remark this answers is not in the document]
Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.Datalocks.DatalocksService.Interfaces/SFA.DAS.Payments.Datalocks.DatalocksService.Interfaces.csproj]: Why would we want to use them in the test?

File: src/SFA.DAS.Payments.Datalocks.Messages/DatalockPaymentEvent.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+﻿using SFA.DAS.Payments.EarningEvents.Messages.Events;
+
+namespace SFA.DAS.Payments.Datalocks.Messages
+{
+    public class DatalockPaymentEvent : ApprenticeshipContractType1EarningEvent , IDatalockPayment
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.Datalocks.Messages/DatalockPaymentEvent.cs]: They aren't actually payments, they are earnings. Can we have two (empty for now) objects:
`PayableEarning`, `NonPayableEarning` inherited from `ApprenticeshipContractType1EarningEvent `?

File: src/SFA.DAS.Payments.Datalocks.DatalocksProxyService/DatalocksProxyService.cs [hunk 4]
```diff
@@ -0,0 +1,33 @@
+﻿using System.Collections.Generic;
+using System.Fabric;
+using Autofac;
+using Microsoft.ServiceFabric.Services.Communication.Runtime;
+using Microsoft.ServiceFabric.Services.Runtime;
+using SFA.DAS.Payments.Application.Infrastructure.Logging;
+using SFA.DAS.Payments.ServiceFabric.Core;
+
+namespace SFA.DAS.Payments.Datalocks.DatalocksProxyService
```
Comment (kuzinator) [inline src/SFA.DAS.Payments.Datalocks.DatalocksProxyService/DatalocksProxyService.cs]: Debatable, but can we have capital "L" in DataLock?

Comment (samuel-dev-dba) [reply src/SFA.DAS.Payments.Datalocks.DatalocksProxyService/DatalocksProxyService.cs]: We can always change that later.

Comment (carlbm) [reply src/SFA.DAS.Payments.Datalocks.DatalocksProxyService/DatalocksProxyService.cs]: V1 uses a lowercase L

**Pull request merged**